### DISCUSSION TITLE
Has the parsing algo changed considerably since 4.0.1?

### DISCUSSION THREAD

**Discussion by geoff-harper** [kaisalmen/WWOBJLoader#79] [Q&A] [answered]
Checking in because I'm facing a strange issue currently with my obj. v4.0.1 and v6.0.0-next are returning different face indices at the exact same 3d point on my mesh. I heavily relied on the way it was in v4.0.1 since it is needed for me to match with face indices in map files for other parts of my system. I never took v5 so I don't know if that's when it changed but based on the release I suspect it was. What should my course of action be?

**Comment by kaisalmen** [comment]
@geoff-harper reminds me I should release a proper version. Did you use `6.0.0-next.1`, because `6.0.0-next.0` is bad.
I first moved the parser code out of OBJLoader2 and then converted the whole code base to TypeScript.
The Parser should have not changed, but it is hard to prove that without unit test. At least, there was no major work / bug-fixes on the Parser part between v4 and v6. Can you provide me files for testing? And can you explain the specific problem in more detail. Thanks.

**Reply by geoff-harper** [reply]
Hey Kai, looks like I'm currently on `6.0.1-next.1`. I don't see the `next` tags on the repo but I took the latest `main` commit on my fork and tried my obj in one of the examples and it's actually returning the triangle that I was seeing on `4.0.1`. Is the `main` branch what's on npm as `6.0.1-next.1`? If I can confirm that then maybe it's back to the drawing board

edit: [here's the branch](https://github.com/geoff-harper/WWOBJLoader/tree/obj-triangle-debug), it's the obj2_basic.html example

**Reply by kaisalmen** [reply]
> edit: [here's the branch](https://github.com/geoff-harper/WWOBJLoader/tree/obj-triangle-debug), it's the obj2_basic.html example

On that branch it is broken?

6.0.1-next.1 was released of this [branch](https://github.com/kaisalmen/WWOBJLoader/tree/next-601), but it does not contain code changes compared to the main.

**Reply by kaisalmen** [reply]
@geoff-harper I have released https://www.npmjs.com/package/wwobjloader2/v/6.0.1-next.2 which is aligned with type definitions of three 156. Can you verify it is really broken with that release and if so please tell me where/how exactly I can that with your model (index example). Thanks

**Reply by geoff-harper** [reply, marked as answer]
I took it and there were no changes to my issue but I have discovered what's causing it. A few weeks ago I added a [bvh to the mesh](https://github.com/gkjohnson/three-mesh-bvh) and hadn't actually compared triangle indices since then. For some reason I never put it together that this is what could be mutating my geometry. Apologies for the rodeo, I should have checked this sooner

**Reply by kaisalmen** [reply]
@geoff-harper no worries. I will release a final 6.0.1 today and will also finally work on the Offscreen Canvas example.

**Reply by geoff-harper** [reply]
Sure I'll pull it when it's up and take a look. I'm noticing some odd behaviour when using it with the react `useLoader` hook but I'll wait until the types are updated before taking a closer look

**Reply by kaisalmen** [reply]
First version of the Offscreen Canvas example is online: 
https://github.com/kaisalmen/WWOBJLoader/blob/main/packages/examples/obj2_basic_offscreen.html
https://github.com/kaisalmen/WWOBJLoader/blob/main/packages/examples/src/examples/OBJLoader2BasicExampleOffscreen.ts

**Reply by kaisalmen** [reply]
Release 6.0.1 is available:
https://www.npmjs.com/package/wwobjloader2/v/6.0.1
